### PR TITLE
support site perf replication

### PR DESCRIPTION
**Pull request by jiuker**
## Description

support site replication

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/support-perf-site-replication.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2022 MinIO, Inc.
```
Comment (harshavardhana) [inline cmd/support-perf-site-replication.go:1]:
```suggestion
// Copyright (c) 2023 MinIO, Inc.
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by shtripat** [comment]
Do we need an active site replication to test the change?

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
> Do we need an active site replication to test the change?

yes you need a `site` replicated setup to test this change.

**Comment by poornas** [comment]
```
➜  mc git:(support_site_replication) mc support perf site-replication sitea --airgap --dev

SiteReplication: ∙●∙

ENDPOINT	RX    	TX
...     	-- MiB	-- MiB
Caught panic:

runtime error: integer divide by zero

Restoring terminal...

goroutine 31 [running]:
runtime/debug.Stack()
	runtime/debug/stack.go:24 +0x64
runtime/debug.PrintStack()
	runtime/debug/stack.go:16 +0x1c
github.com/charmbracelet/bubbletea.(*Program).Run.func1()
	github.com/charmbracelet/bubbletea@v0.23.2/tea.go:402 +0x8c
panic({0x10131f560, 0x101c65720})
	runtime/panic.go:890 +0x258
github.com/minio/mc/cmd.(*speedTestUI).View(0x1400052e790)
	github.com/minio/mc/cmd/speedtest-spinner.go:255 +0x280c
github.com/charmbracelet/bubbletea.(*Program).eventLoop(0x1400052e840, {0x10149c0d0?, 0x1400052e790?}, 0x0?)
	github.com/charmbracelet/bubbletea@v0.23.2/tea.go:345 +0x5c8
github.com/charmbracelet/bubbletea.(*Program).Run(0x1400052e840)
	github.com/charmbracelet/bubbletea@v0.23.2/tea.go:465 +0x6b0
github.com/minio/mc/cmd.mainAdminSpeedTestSiteReplications.func2()
	github.com/minio/mc/cmd/support-perf-site-replication.go:86 +0x28
created by github.com/minio/mc/cmd.mainAdminSpeedTestSiteReplications
	github.com/minio/mc/cmd/support-perf-site-replication.go:85 +0x508
mc: MinIO performance report saved at sitea-perf_20230705175628.zip, please upload to SUBNET portal manually
```
@jiuker , there's a panic here

**Review by poornas — APPROVED** [review]

File: cmd/support-perf.go [hunk 2]
```diff
@@ -112,10 +113,11 @@ EXAMPLES:
 
 // PerfTestOutput - stores the final output of performance test(s)
 type PerfTestOutput struct {
-	ObjectResults *ObjTestResults   `json:"object,omitempty"`
-	NetResults    *NetTestResults   `json:"network,omitempty"`
-	DriveResults  *DriveTestResults `json:"drive,omitempty"`
-	Error         string            `json:"error,omitempty"`
+	ObjectResults          *ObjTestResults             `json:"object,omitempty"`
+	NetResults             *NetTestResults             `json:"network,omitempty"`
+	SiteReplicationResults *SiteReplicationTestResults `json:"site_replication,omitempty"`
```
Comment (harshavardhana) [inline cmd/support-perf.go:118]:
```suggestion
	SiteReplicationResults *SiteReplicationTestResults `json:"siteReplication,omitempty"`
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
Please fix the conflicts @jiuker by rebasing with master

File: cmd/support-perf-site-replication.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	"os"
+	"time"
+
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/minio/cli"
+	"github.com/minio/madmin-go/v3"
+	"github.com/minio/mc/pkg/probe"
+)
+
+func mainAdminSpeedTestSiteReplications(ctx *cli.Context, aliasedURL string, outCh chan<- PerfTestResult) error {
```
Comment (harshavardhana) [inline cmd/support-perf-site-replication.go:31]:
```suggestion
func mainAdminSpeedTestSiteReplication(ctx *cli.Context, aliasedURL string, outCh chan<- PerfTestResult) error {
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

File: cmd/support-perf.go [hunk 4]
```diff
@@ -458,6 +507,8 @@ func runPerfTests(ctx *cli.Context, aliasedURL string, perfType string) []PerfTe
 			mainAdminSpeedTestObject(ctx, aliasedURL, resultCh)
 		case "net":
 			mainAdminSpeedTestNetperf(ctx, aliasedURL, resultCh)
+		case "site-replication":
+			mainAdminSpeedTestSiteReplications(ctx, aliasedURL, resultCh)
```
Comment (harshavardhana) [inline cmd/support-perf.go:511]:
```suggestion
			mainAdminSpeedTestSiteReplication(ctx, aliasedURL, resultCh)
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**